composition-js/src/__tests__/compose.test.ts

```diff
@@ -4008,4 +4009,287 @@ describe('composition', () => {
       assertCompositionSuccess(result);
     });
   });
+
+  describe('@authenticated', () => {
+    it('comprehensive locations', () => {
```

Comment:
Just occurred to me, but it's probably worth having a case in this test or a separate test for `@interfaceObject`. That if, if you have:
```graphql
type Foo @interfaceObject @authenticated {
   ...
}
```
then we want to make sure the `@authenticated` gets added to the `Foo` interface in the supergraph (I suspect/hope it works, but not sure we have existing tests for propagation of directive on interface object types to their corresponding interface, so worth adding one here).

Reply:
Good catch! It works.